WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCoupon.swift

```diff
@@ -17,48 +17,55 @@ struct AddEditCoupon: View {
                                 Divider()
                                     .padding(.leading, Constants.margin)
+
+                                Text(viewModel.amountSubtitleText)
+                                    .subheadlineStyle()
+                                    .padding(.horizontal, Constants.margin)
                             }
+                            .padding([.bottom], Constants.verticalSpacing)
 
-                            Text(Localization.footerCouponAmountPercentage)
-                                .subheadlineStyle()
-                                .padding(.horizontal, Constants.margin)
 
                             Group {
                                 TitleAndTextFieldRow(title: Localization.couponCode,
-                                                     placeholder: Localization.couponCode,
+                                                     placeholder: Localization.couponCodePlaceholder,
                                                      text: $viewModel.codeField,
-                                                     editable: false,
+                                                     editable: true,
                                                      fieldAlignment: .leading,
-                                                     keyboardType: .asciiCapableNumberPad)
+                                                     keyboardType: .default)
                                 Divider()
                                     .padding(.leading, Constants.margin)
+                                    .padding([.bottom], Constants.verticalSpacing)
+                                Text(Localization.footerCouponCode)
+                                    .subheadlineStyle()
+                                    .padding(.horizontal, Constants.margin)
                             }
-
-                            Text(Localization.footerCouponCode)
-                                .subheadlineStyle()
-                                .padding(.horizontal, Constants.margin)
-
-                            //TODO: leading aligning for this button
-                            Button {
-                                //TODO: handle action
-                            } label: {
-                                Text(Localization.regenerateCouponCodeButton)
+                            .padding([.bottom], Constants.verticalSpacing)
+
+
+                            HStack {
+                                Button {
+                                    //TODO: handle action
+                                } label: {
+                                    Text(Localization.regenerateCouponCodeButton)
+                                }
+                                .buttonStyle(LinkButtonStyle())
+                                .fixedSize()
+                                Spacer().frame(maxWidth: .infinity)
                             }
```

Comment:
Suggestion: this HStack can be replaced with the `.frame` modifier and `.leading` alignment:
```suggestion
                                Button {
                                    //TODO: handle action
                                } label: {
                                    Text(Localization.regenerateCouponCodeButton)
                                }
                                .buttonStyle(LinkButtonStyle())
                                .fixedSize()
                                .frame(maxWidth: .infinity, alignment: .leading)
```

Reply:
One thing I haven't figured out though - there seems to be a tiny spacing on the leading side of the button title. Maybe it's something that the button does:

![Untitled](https://user-images.githubusercontent.com/5533851/161885411-1ced333e-761f-45e2-949f-ae28b0e11169.jpg)

Reply:
> One thing I haven't figured out though - there seems to be a tiny spacing on the leading side of the button title. Maybe it's something that the button does.

It looks like all our Button Style are using a `defaultEdgeInsets` configured like that: `EdgeInsets(top: 12, leading: 22, bottom: 12, trailing: 22)`. Do you believe it's worth overriding the configuration just for this use case?

Reply:
Great find! It would be nice to add the option to override the edges if it doesn't require too much effort, but totally not mandatory I think!